2.6/contrib/common.sh

```diff
@@ -0,0 +1,167 @@
+function no_endpoints() {
+  [ -z "$(endpoints)" ] && return 0
+  [ "$(endpoints)" == "$(container_addr)" ]
+}
+
+# build_mongo_config builds the MongoDB replicaSet config used for the cluster
+# initialization
+function build_mongo_config() {
+  local members="{ _id: 0, host: \"$(mongo_addr)\"},"
+  local member_id=1
+  local container_addr="$(container_addr)"
+  for node in $(endpoints); do
+    [ "$node" == container_addr ] && continue
+    members+="{ _id: ${member_id}, host: \"${node}:${CONTAINER_PORT}\"},"
+    let member_id++
+  done
+  echo -n "var config={ _id: \"${MONGODB_REPLICA_NAME}\", members: [ ${members%,} ] }"
+}
+
+# mongo_initiate initiate the replica set
+function mongo_initiate() {
+  local mongo_wait="while (rs.status().startupStatus || (rs.status().hasOwnProperty(\"myState\") && rs.status().myState != 1)) { printjson( rs.status() ); sleep(1000); }; printjson( rs.status() );"
+  config=$(build_mongo_config)
+  echo "=> Initiating MongoDB replica using: ${config}"
+  mongo admin --eval "${config};rs.initiate(config);${mongo_wait}"
+}
+
+# get the address of the current primary member
+function mongo_primary_member_addr() {
+  local current_endpoints=$(endpoints)
+  local mongo_node="$(echo -n ${current_endpoints} | cut -d ' ' -f 1):${CONTAINER_PORT}"
+  echo -n $(mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host ${mongo_node} --quiet --eval "print(rs.isMaster().primary);")
+}
+
+# mongo_remove removes the current MongoDB from the cluster
+function mongo_remove() {
+  echo "=> Removing $(mongo_addr) on $(mongo_primary_member_addr) ..."
+  mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" \
+    --host $(mongo_primary_member_addr) --eval "rs.remove('$(mongo_addr)');" &>/dev/null || true
+}
+
+# mongo_add advertise the current container to other mongo replicas
+function mongo_add() {
+  echo "=> Adding $(mongo_addr) to $(mongo_primary_member_addr) ..."
+  mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" \
+    --host $(mongo_primary_member_addr) --eval "rs.add('$(mongo_addr)');"
+}
+
+# run_mongod_supervisor runs the MongoDB replica supervisor that manages
+# registration of the new members to the MongoDB replica cluster
+function run_mongod_supervisor() {
+  local log_pipe_path=$(mktemp --suffix=log)
+  rm -rf ${log_pipe_path}
+  ( mkfifo ${log_pipe_path} && cat ${log_pipe_path} ) &
+  /var/lib/mongodb/replica_supervisor.sh ${log_pipe_path} &
+}
+
+# mongo_create_users creates the MongoDB admin user and the database user
+# configured by MONGO_USER
+function mongo_create_users() {
+  mongo admin --eval "db.createUser({user: 'admin', pwd: '${MONGODB_ADMIN_PASSWORD}', roles: ['dbAdminAnyDatabase', 'userAdminAnyDatabase' , 'readWriteAnyDatabase','clusterAdmin' ]})"
+
+  mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --eval "db.getSiblingDB('${MONGODB_DATABASE}').createUser({user: '${MONGODB_USER}', pwd: '${MONGODB_PASSWORD}', roles: [ 'readWrite' ]});"
```

Comment:
Simple replacing is not possible. It is because of a change in "localhost exeption". In mongodb 2.6 it is only active till first user is created (in 2.4 it was active all the time):

>Localhost Exception
>
>The localhost exception allows you to enable authorization before creating the first user in the system. When active, the localhost exception allows all connections from the localhost interface to have full access to that instance. The exception applies only when there are no users created in the MongoDB instance.

https://docs.mongodb.org/v2.6/core/authentication/

This is a problem during creating users in replication. In mongodb 2.6 setting keyfile implies authorization (http://docs.mongodb.org/v2.6/reference/configuration-options/#security.keyFile). So auth is on and "normal user" it not allowed to create new users. So admin user have to be created first...